api_app/script_analyzers/observable_analyzers/zoomeye.py

```diff
@@ -0,0 +1,107 @@
+import requests
+
+from api_app.exceptions import AnalyzerRunException, AnalyzerConfigurationException
+from api_app.script_analyzers import classes
+from intel_owl import secrets
+
+
+class ZoomEye(classes.ObservableAnalyzer):
+    base_url: str = "https://api.zoomeye.org/"
+
+    def set_config(self, additional_config_params):
+        self.api_key_name = additional_config_params.get("api_key_name", "ZOOMEYE_KEY")
+        self.search_type = additional_config_params.get("search_type", "host")
+
+        # Setting query filters
+        self.query = {}
+        host_search_filters = [
+            "app",
+            "ver",
+            "device",
+            "os",
+            "service",
+            "cidr",
+            "port",
+            "city",
+            "country",
+            "asn",
+        ]
+        web_search_filters = [
+            "header",
+            "keywords",
+            "desc",
+            "title",
+            "site",
+        ]  # Included only which aren't included above
+        if "query" in additional_config_params:
+            for filter in host_search_filters:
+                self.query[filter] = additional_config_params["query"].get(filter, None)
+            for filter in web_search_filters:
+                self.query[filter] = additional_config_params["query"].get(filter, None)
+
+        self.pages = additional_config_params.get("page", None)
```

Comment:
shouldn't this be `self.page` ?

Reply:
Either works, Because, While preparing URL, I am passing `page`. But Yes, using `page` instead of `pages` seems more appropriate